src/NuGet.Clients/NuGet.Credentials/AuthTypeFilteredCredentials.cs

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Linq;
+
+
+namespace NuGet.Credentials
+{
+    /// <summary>
+    /// Wraps another ICredentials object by returning null if the authType is not in a specified allow list
+    /// </summary>
+    public class AuthTypeFilteredCredentials : ICredentials
+    {
+        private readonly string[] _authTypeFilter;
+        private readonly ICredentials _innerCredential;
+
+        /// <summary>
+        /// Initializes a new AuthTypeFilteredCredentials
+        /// </summary>
+        /// <param name="innerCredential">Credential to delegate to</param>
+        /// <param name="authTypeFilter">List of authTypes to respond to. May be null, in which case all auth types are allowed.</param>
+        public AuthTypeFilteredCredentials(ICredentials innerCredential, IEnumerable<string> authTypeFilter)
```

Comment:
Do not allow empty or null authTypes, - one can use regular credentials object instead.